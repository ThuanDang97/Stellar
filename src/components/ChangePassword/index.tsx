import React, { memo } from 'react'
import { useTheme } from 'styled-components'

// Components
import { BoxShadowStyled, TextStyled } from '@components/styled-components'
import Title from '@components/Title'
import LinkComponent from '@components/Link'

// Utils
import { pxToRem } from '@utils/theme'

// Constants
import { DEFAULT_HEADER_URL } from '@constants/routes'

const ChangePassword = () => {
  const theme = useTheme()

  return (
    <BoxShadowStyled
      display="flex"
      flexDirection="column"
      width={theme.metrics.width.full}
      padding={`${pxToRem(theme.metrics.dimensions.lg)} ${pxToRem(
        theme.metrics.dimensions.xlg,
      )}`}
    >
      <Title title="Password changed!" color={theme.colors.black} />

      <TextStyled
        mTop={theme.metrics.dimensions.md}
        mBottom={theme.metrics.dimensions.xmd}
      >
        Your password has been changed successfully
      </TextStyled>

      <LinkComponent
        href={DEFAULT_HEADER_URL.LOGIN.URL}
        linkTypes="unLockLink"
        text="Okay"
        pTop={theme.metrics.dimensions.sm}
        pBottom={theme.metrics.dimensions.sm}
      />
    </BoxShadowStyled>
  )
}

export default memo(ChangePassword)
