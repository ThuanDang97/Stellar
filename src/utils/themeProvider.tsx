import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
