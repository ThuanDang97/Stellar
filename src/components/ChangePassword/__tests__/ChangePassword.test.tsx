// Components
import ChangePassword from '@components/ChangePassword'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

describe('ChangePassword component', () => {
  it('should renders ChangePassword unchanged', () => {
    const { asFragment } = renderWithTheme(<ChangePassword />)

    expect(asFragment()).toMatchSnapshot()
  })
})
