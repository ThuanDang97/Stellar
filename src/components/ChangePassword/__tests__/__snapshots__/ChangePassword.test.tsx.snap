// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword component should renders ChangePassword unchanged 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 qDPxs iHqAbY"
    display="flex"
    width="100%"
  >
    <h1
      class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 bfcxuE WstJK"
      color="#000"
    >
      Password changed!
    </h1>
    <span
      class="TextStyled-sc-9xia41-0 cQkIvB"
    >
      Your password has been changed successfully
    </span>
    <a
      class="LinkStyled__ButtonLinkStyled-sc-626068-1 dlbvFT"
      data-testid="link"
      href="/login"
    >
      Okay
    </a>
  </div>
</DocumentFragment>
`;
